python/tests/table/test_metadata.py

```diff
@@ -0,0 +1,328 @@
+            "schema-id": 0,
+        },
+    ],
+    "snapshot-log": [
+        {"timestamp-ms": 1637943123188, "snapshot-id": 7681945274687743099},
+    ],
+    "metadata-log": [
+        {
+            "timestamp-ms": 1637943123331,
+            "metadata-file": "3://foo/bar/baz/00000-907830f8-1a92-4944-965a-ff82c890e912.metadata.json",
+        }
+    ],
+}
+EXAMPLE_TABLE_METADATA_V2 = {
+    "format-version": 2,
+    "table-uuid": "foo-table-uuid",
+    "location": "s3://foo/bar/baz.metadata.json",
+    "last-updated-ms": 1600000000000,
+    "last-column-id": 4,
+    "last-sequence-number": 1,
+    "schemas": [
+        {
+            "schema-id": 0,
+            "fields": [
+                {"id": 1, "name": "foo", "required": True, "type": "string"},
+                {"id": 2, "name": "bar", "required": True, "type": "string"},
+                {"id": 3, "name": "baz", "required": True, "type": "string"},
+                {"id": 4, "name": "qux", "required": True, "type": "string"},
+            ],
+            "identifier-field-ids": [],
+        }
+    ],
+    "current-schema-id": 0,
+    "default-spec-id": 0,
+    "partition-specs": [{"spec-id": 0, "fields": []}],
+    "last-partition-id": 999,
+    "default-sort-order-id": 0,
+    "sort-orders": [{"order-id": 0, "fields": []}],
+    "properties": {"owner": "root", "write.format.default": "parquet", "read.split.target.size": 134217728},
+    "current-snapshot-id": 7681945274687743099,
+    "snapshots": [
+        {
+            "snapshot-id": 7681945274687743099,
+            "timestamp-ms": 1637943123188,
+            "summary": {
+                "operation": "append",
+                "added-data-files": "6",
+                "added-records": "237993",
+                "added-files-size": "3386901",
+                "changed-partition-count": "1",
+                "total-records": "237993",
+                "total-files-size": "3386901",
+                "total-data-files": "6",
+                "total-delete-files": "0",
+                "total-position-deletes": "0",
+                "total-equality-deletes": "0",
+            },
+            "manifest-list": "s3://foo/bar/baz/snap-2874264644797652805-1-9cb3c3cf-5a04-40c1-bdd9-d8d7e38cd8e3.avro",
+            "schema-id": 0,
+        },
+    ],
+    "snapshot-log": [
+        {"timestamp-ms": 1637943123188, "snapshot-id": 7681945274687743099},
+    ],
+    "metadata-log": [
+        {
+            "timestamp-ms": 1637943123331,
+            "metadata-file": "3://foo/bar/baz/00000-907830f8-1a92-4944-965a-ff82c890e912.metadata.json",
+        }
+    ],
+}
+
+
+@pytest.mark.parametrize(
+    "metadata",
+    [
+        EXAMPLE_TABLE_METADATA_V1,
+        EXAMPLE_TABLE_METADATA_V2,
+    ],
+)
+def test_from_dict(metadata: dict):
+    """Test initialization of a TableMetadata instance from a dictionary"""
+    TableMetadata.parse_obj(metadata)
```

Comment:
Is this just testing that an object is produced? Shouldn't it validate the contents?

Reply:
This is just checking the logic. This will be obsolete once https://github.com/samuelcolvin/pydantic/pull/3847 has been merged :)